container/network/tcp.go

```diff
@@ -9,23 +9,89 @@ import (
 	"strconv"
 	"strings"
 
-	"github.com/intelsdi-x/snap-plugin-collector-docker/mounts"
-	"github.com/intelsdi-x/snap-plugin-collector-docker/wrapper"
+	"github.com/intelsdi-x/snap-plugin-collector-docker/container"
+
+	log "github.com/Sirupsen/logrus"
 )
 
+type Tcp struct{}
+
+func (t *Tcp) GetStats(stats *container.Statistics, opts container.GetStatOpt) error {
+	pid, err := opts.GetIntValue("pid")
+	if err != nil {
+		return err
+	}
+
+	isHost, err := opts.GetBoolValue("is_host")
+	if err != nil {
+		return err
+	}
+
+	procfs, err := opts.GetStringValue("procfs")
+	if err != nil {
+		return err
+	}
+
+	if !isHost {
+		path := filepath.Join(procfs, strconv.Itoa(pid))
+		stats.Connection.Tcp, err = TcpStatsFromProc(path)
+		if err != nil {
+			// only log error message
+			log.WithFields(log.Fields{
+				"module": "network",
+				"block":  "GetStats",
+			}).Errorf("Unable to get network stats, pid %d: %s", pid, err)
+		}
+	}
+
+	return nil
+}
```

Comment:
Notice that tcp metrics and tcp6 metrics are exactly the same. what is more the way of retrieving these data differs only in the taken path. For this refactored code distinguishing between tcp and tcp is made on defined structures, please consider putting all this code from GetStats in different function with input argument pointing to "net/tcp" or "net/tcp6" depending on a case and use `tcpStatsFromProc` directly, so functions `TcpStatsFromProc` and `Tcp6StatsFromProc` are not longer needed.